### PR TITLE
Animate Menu and Modal

### PR DESCRIPTION
**Pull request by vitokhangnguyen**
Just checking out random animations stuff and ended up here

### CODE REVIEW

File: src/ui/Modal/Modal.tsx [hunk 1]
```diff
@@ -33,36 +37,73 @@ const Modal: FC<ModalProps> = ({
...
+          initial="hidden"
+          animate="visible"
+          exit="hidden"
+        >
+          <AnimatedReachDialogContent
+            id={modalId}
+            aria-labelledby={modalTitleId}
+            css={modalDialogCss(size)}
+            {...props}
+            transition={{ duration: animationDuration }}
```
Comment (Andrewnt219) [inline src/ui/Modal/Modal.tsx:56]: Nit: should use `spring` animation, it's more realistic. Using `duration` set animation to `tween`.

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]

File: src/ui/Menu/Menu.tsx [hunk 2]
```diff
@@ -29,26 +33,63 @@ type MenuProps = RadixDropdownMenu.DropdownMenuProps & {
...
+ */
+const animationVariants: Variants = {
+  visible: {
+    scale: 1,
+  },
+  hidden: {
+    scale: 0,
+  },
+};
+
```
[the remark this answers is not in the document]
Comment (vitokhangnguyen) [reply src/ui/Menu/Menu.tsx]: Just changed it to animating `height`

File: src/ui/Menu/Menu.tsx [hunk 3]
```diff
@@ -29,26 +33,68 @@ type MenuProps = RadixDropdownMenu.DropdownMenuProps & {
...
+  duration: 0.5,
+};
+const ANIMATION_VARIANTS: Variants = {
+  visible: {
+    height: 'auto',
+  },
+  hidden: {
+    height: 0,
+  },
+};
```
Comment (Andrewnt219) [inline src/ui/Menu/Menu.tsx:92]: this is a cool animation. Don't abuse animating height though, it's [costly](https://web.dev/animations-guide/#triggers).

**Review by Andrewnt219 — APPROVED** [review]

File: src/ui/Menu/Menu.tsx (hunk 3 above)
Comment (vitokhangnguyen) [reply src/ui/Menu/Menu.tsx:92]: Yea, I understand that `height` is a CPU-calculated parameter while `transform` is GPU-calculated so it's more heavy. But what're the alternative here for this animation?

Comment (vitokhangnguyen) [reply src/ui/Menu/Menu.tsx:92]: I guess just don't use it on animation that may happen very frequently?

Comment (Andrewnt219) [reply src/ui/Menu/Menu.tsx:92]: To make the same animation? Not possible. Just don't use these kinds of animation too much, especially on mobile.

**Pull request merged**